Add TypeScript Guides (#6454)
DevExpress/devextreme-documentation: concepts/Common/TypeScript Guides/15 Events/05 Event Objects.md

> Each component has event types that define arguments for handling events. Naming convention is the following: the argument type of the `contentReady` event handler is `ContentReadyEvent`.

Reviewer: ```suggestion
DevExtreme components define types that supply event argument data. Naming convention is the following: the argument type of the `contentReady` event handler is `ContentReadyEvent`.
```

> Events are unions built up from [common and specific parts](/Documentation/Guide/Common/TypeScript_Guides/Events/#Writable_Fields) (to simplify reading them):

Reviewer: I'm not sure I'm 100% correct and let me know if I'm not. But the phrase "common and specific parts" is too vague. 

```suggestion
Events are unions that build [specific implementation on top of basic event types](/Documentation/Guide/Common/TypeScript_Guides/Events/#Writable_Fields). This implementation makes event definitions easily readable:
```

> Event objects are unions built up from [basic event types](/Documentation/Guide/Common/TypeScript_Guides/Events/#Event_Object_Parts). This makes event definitions easily readable:

Reviewer 2: ```suggestion
Event objects are unions created from [basic event types](/Documentation/Guide/Common/TypeScript_Guides/Events/#Event_Object_Parts). This improves the readability of event definitions:
```
